control_toolbox/include/control_toolbox/low_pass_filter.hpp

```diff
@@ -123,10 +124,17 @@ class LowPassFilter
   // Filter parameters
   double a1_; /** feedbackward coefficient. */
   double b1_; /** feedforward coefficient. */
-  /** internal data storage of template type. */
-  T filtered_value, filtered_old_value, old_value;
-  /** internal data storage (wrench). */
-  Eigen::Matrix<double, 6, 1> msg_filtered, msg_filtered_old, msg_old;
+
+  // Define the storage type based on T
+  using StorageType = typename std::conditional<
+    std::is_same<T, geometry_msgs::msg::WrenchStamped>::value, Eigen::Matrix<double, 6, 1>,
```

Comment:
```suggestion
    std::is_same<T, geometry_msgs::msg::WrenchStamped>::value, Eigen::Vector6d,
```
May be we could use this directly

Reply:
This might come from 
https://github.com/ros/geometry/blob/noetic-devel/eigen_conversions/include/eigen_conversions/eigen_msg.h#L100
according to the comment
`  // TODO(destogl): use wrenchMsgToEigen`
But I don't find these conversion wrappers for ROS 2, https://docs.ros.org/en/rolling/p/tf2_eigen/generated/index.html#tf2-eigen